app/web/src/store/realtime/heimdall.ts

```diff
@@ -182,16 +182,17 @@ const updateCache = (
   if (!removed && !data) return;
 
   queryClient.setQueryData(queryKey, (cachedData: { id: string }[]) => {
-    // Don't mutate if we haven't got any data yet
-    // TODO it seems wrong not to add items to an empty list ...
-    if (!cachedData || cachedData.length === 0) {
-      return cachedData;
+    if (!cachedData) {
+      return;
```

Comment:
This needs to return cachedData, right? That's at least how their examples worked--if you don't modify it you return it. (I guess the code you are writing would return undefined, but the intent is "leave it unchanged" rather than "set it to undefined.")

Reply:
sure.. in this case `cachedData` _is_ undefined.. but i can be more explicit.